terraform/03_gke_cluster.tf

```diff
@@ -146,13 +146,33 @@ resource "null_resource" "install_istio" {
   depends_on = [null_resource.set_gke_context]
 }
 
+# Get GKE cluster's default service credentials through project_number
+# Solves Issue #372. When upstream bug is resolved, remove this.
+resource "null_resource" "get_credentials" {
+  provisioner "local-exec" {
+    command = "gcloud iam service-accounts keys create key.json --iam-account $(gcloud projects list --filter='${data.google_project.project.project_id}' --format='value(PROJECT_NUMBER)')-compute@developer.gserviceaccount.com"
+  }
+
+  depends_on = [null_resource.install_istio]
+}
+
+# Forwards credentials as a secret called "my-key"
+# Accessible to containers via GOOGLE_APP_CREDENTIALS env variable.
+resource "null_resource" "forward_credentials" {
+  provisioner "local-exec" {
+    command = "kubectl create secret generic my-key --from-file=key.json=key.json"
```

Comment:
similar note on using terraform provider for k8s if possible

Reply:
Noted on the above. Yes, it looks like I can set a "kubernetes_secret" to do this instead. Similarly with Google IAM.